mautic/mautic
# Tracking lead manipulator

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | N
| New feature? | Y
| Automated tests included? |N
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | 
| BC breaks? | N
| Deprecations? | N

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
Log source of lead creation and identification

[//]: # ( As applicable: )
#### Steps to test this PR:
1. Run migrations `php app/console doctrine:migrations:migrate`
2. Create a new lead through the following sources (clear your browsers cache each time): tracking pixel on 3rd party page and via a Mautic landing page. Check the contacts created and note the created by source entry in the history should be the URL for the tracking pixel and the name of the landing page for the other. 
3. Now submit a Mautic form embedded in a 3rd party page then also a Mautic form standalone (/form/ID). Note the identified by source entry should be the name of the form in the timeline. 
4. Do a CRM sync and note the created by and identified source should be the CRM name
5. Do an import and then also manually create a contact and the created by/identified by source should be the name of the logged in user.
